### PR TITLE
storage: add topic column to mz_kafka_sources

### LINKED ISSUE #24996: Add `topic` to `mz_kafka_sources` system catalog table
### Feature request

In the same way that we make `topic` available in `mz_kafka_sinks`, we should also make this metadata available in `mz_sources`. 

Requested by @RobinClowers.

### PR DESCRIPTION
**Pull request by guswynn**
Closes #24996

The fingerprint of the table is changing, so this change is backward compatible.

### Motivation

  * This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - add `topic` column to `mz_internal.mz_kafka_sources`

### CODE REVIEW

**Review by umanwizard — APPROVED** [review]

**Review by benesch — APPROVED** [review]
Huge thank you for just knocking this one out. Don't forget a release note!

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 1]
```diff
@@ -336,6 +336,7 @@ The `mz_kafka_sources` table contains a row for each Kafka source in the system.
 |------------------------|----------------|-----------------------------------------------------------------------------------------------------------|
 | `id`                   | [`text`]       | The ID of the Kafka source. Corresponds to [`mz_catalog.mz_sources.id`](../mz_catalog#mz_sources).        |
 | `group_id_prefix`      | [`text`]       | The value of the `GROUP ID PREFIX` connection option.                                                     |
+| `topic          `      | [`text`]       | The topic this Kafka source is reading from.                                                              |
```
Comment (morsapaes) [inline doc/user/content/sql/system-catalog/mz_internal.md:339]:
```suggestion
| `topic          `      | [`text`]       | The name of the Kafka topic the source is reading from.                                                              |
```

Comment (morsapaes) [reply doc/user/content/sql/system-catalog/mz_internal.md:339]: 🫠 nit. For consistency with the description in `mz_kafka_sinks`.

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 1 above)
Comment (guswynn) [reply doc/user/content/sql/system-catalog/mz_internal.md:339]: rip

**Pull request merged**